plan/validator.go

```diff
@@ -302,3 +307,49 @@ func checkDuplicateColumnName(indexColNames []*ast.IndexColName) error {
 	}
 	return nil
 }
+
+// checkFieldLengthLimitation check the maximum length of the column
+// See https://dev.mysql.com/doc/refman/5.7/en/storage-requirements.html
+func checkFieldLengthLimitation(colDef *ast.ColumnDef) error {
+	tp := colDef.Tp
+	if tp == nil {
+		return nil
+	}
+	if tp.Flen > mysql.MaxUint32 {
+		return errors.Errorf("Display width out of range for column '%s' (max = %d)", colDef.Name.Name.O, mysql.MaxUint32)
```

Comment:
return terror to return correct mysql error code, please see [util/types/errors.go](https://github.com/pingcap/tidb/blob/master/util/types/errors.go#L27), you should return error like `types.ErrOverflow`, which will carry errCode